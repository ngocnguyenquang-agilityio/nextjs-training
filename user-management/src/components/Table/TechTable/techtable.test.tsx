import React, { ReactElement } from 'react';
import { render, fireEvent } from '@testing-library/react';

// Components
import { TechTable } from '.';

describe('TechTable Component', () => {
  jest.mock(
    'next/link',
    () =>
      ({ children, ...rest }: { children: ReactElement }) =>
        React.cloneElement(children, { ...rest })
  );

  test('Should render successfully', () => {
    const comp = render(<TechTable />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<TechTable />);
    expect(comp).toMatchSnapshot();
  });

  test('Should click delete button', async () => {
    const comp = render(<TechTable />);
    const alertMock = jest.spyOn(window, 'alert').mockImplementation();
    const button = comp.queryByTestId('delete-5') as HTMLButtonElement;
    await fireEvent.click(button);
    expect(alertMock.mock.calls.length).toBe(1);
  });

  test('Should able to click correct edit button', async () => {
    const comp = render(<TechTable />);
    const button = comp.queryByTestId('edit-5') as HTMLLinkElement;
    await fireEvent.click(button);
    expect(button.getAttribute('href')).toBe('/edit-tech/5');
  });
});
