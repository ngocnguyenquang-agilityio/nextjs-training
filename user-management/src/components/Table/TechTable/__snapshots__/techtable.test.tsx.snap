// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechTable Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="w-full text-sm text-left text-gray-500"
      >
        <thead
          class="text-xs text-gray-700 uppercase bg-gray-50"
        >
          <tr>
            <th
              class="px-6 py-3"
              scope="col"
            >
              Name
            </th>
            <th
              class="px-6 py-3"
              scope="col"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          class="bg-white"
        >
          <tr
            class="odd:bg-white even:bg-gray-50 border-b hover:bg-gray-100"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <img
                  alt="TypeScript logo"
                  class="rounded-full"
                  data-nimg="1"
                  decoding="async"
                  height="28"
                  loading="lazy"
                  src="https://symbols.getvecta.com/stencil_25/87_typescript.cb2d7326fa.svg"
                  style="color: transparent;"
                  width="28"
                />
                <p>
                  TypeScript
                </p>
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <a
                  class="group rounded-md border p-2 hover:bg-blue-400"
                  data-testid="edit-4"
                  href="/edit-tech/4"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 group-hover:text-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L6.832 19.82a4.5 4.5 0 01-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 011.13-1.897L16.863 4.487zm0 0L19.5 7.125"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </a>
                <button
                  class="flex items-center text-sm font-md bg-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-2 py-1 rounded-md false group bg-transparent rounded-md border hover:bg-red-400"
                  data-testid="delete-4"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 group-hover:text-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50 border-b hover:bg-gray-100"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <img
                  alt="React logo"
                  class="rounded-full"
                  data-nimg="1"
                  decoding="async"
                  height="28"
                  loading="lazy"
                  src="https://symbols.getvecta.com/stencil_25/72_react.76a8d36b4b.svg"
                  style="color: transparent;"
                  width="28"
                />
                <p>
                  React
                </p>
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <a
                  class="group rounded-md border p-2 hover:bg-blue-400"
                  data-testid="edit-5"
                  href="/edit-tech/5"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 group-hover:text-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L6.832 19.82a4.5 4.5 0 01-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 011.13-1.897L16.863 4.487zm0 0L19.5 7.125"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </a>
                <button
                  class="flex items-center text-sm font-md bg-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-2 py-1 rounded-md false group bg-transparent rounded-md border hover:bg-red-400"
                  data-testid="delete-5"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 group-hover:text-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50 border-b hover:bg-gray-100"
          >
            <td
              class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
            >
              <div
                class="flex items-center gap-3"
              >
                <img
                  alt="NextJS logo"
                  class="rounded-full"
                  data-nimg="1"
                  decoding="async"
                  height="28"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Fstatic-00.iconduck.com%2Fassets.00%2Fnext-js-icon-2048x2048-5dqjgeku.png&w=64&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fstatic-00.iconduck.com%2Fassets.00%2Fnext-js-icon-2048x2048-5dqjgeku.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fstatic-00.iconduck.com%2Fassets.00%2Fnext-js-icon-2048x2048-5dqjgeku.png&w=64&q=75 2x"
                  style="color: transparent;"
                  width="28"
                />
                <p>
                  NextJS
                </p>
              </div>
            </td>
            <td
              class="px-6 py-4"
            >
              <div
                class="flex gap-3"
              >
                <a
                  class="group rounded-md border p-2 hover:bg-blue-400"
                  data-testid="edit-6"
                  href="/edit-tech/6"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 group-hover:text-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L6.832 19.82a4.5 4.5 0 01-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 011.13-1.897L16.863 4.487zm0 0L19.5 7.125"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </a>
                <button
                  class="flex items-center text-sm font-md bg-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-2 py-1 rounded-md false group bg-transparent rounded-md border hover:bg-red-400"
                  data-testid="delete-6"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 group-hover:text-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="w-full text-sm text-left text-gray-500"
    >
      <thead
        class="text-xs text-gray-700 uppercase bg-gray-50"
      >
        <tr>
          <th
            class="px-6 py-3"
            scope="col"
          >
            Name
          </th>
          <th
            class="px-6 py-3"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        class="bg-white"
      >
        <tr
          class="odd:bg-white even:bg-gray-50 border-b hover:bg-gray-100"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <img
                alt="TypeScript logo"
                class="rounded-full"
                data-nimg="1"
                decoding="async"
                height="28"
                loading="lazy"
                src="https://symbols.getvecta.com/stencil_25/87_typescript.cb2d7326fa.svg"
                style="color: transparent;"
                width="28"
              />
              <p>
                TypeScript
              </p>
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <a
                class="group rounded-md border p-2 hover:bg-blue-400"
                data-testid="edit-4"
                href="/edit-tech/4"
              >
                <svg
                  aria-hidden="true"
                  class="w-5 group-hover:text-white"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L6.832 19.82a4.5 4.5 0 01-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 011.13-1.897L16.863 4.487zm0 0L19.5 7.125"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </a>
              <button
                class="flex items-center text-sm font-md bg-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-2 py-1 rounded-md false group bg-transparent rounded-md border hover:bg-red-400"
                data-testid="delete-4"
              >
                <svg
                  aria-hidden="true"
                  class="w-5 group-hover:text-white"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50 border-b hover:bg-gray-100"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <img
                alt="React logo"
                class="rounded-full"
                data-nimg="1"
                decoding="async"
                height="28"
                loading="lazy"
                src="https://symbols.getvecta.com/stencil_25/72_react.76a8d36b4b.svg"
                style="color: transparent;"
                width="28"
              />
              <p>
                React
              </p>
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <a
                class="group rounded-md border p-2 hover:bg-blue-400"
                data-testid="edit-5"
                href="/edit-tech/5"
              >
                <svg
                  aria-hidden="true"
                  class="w-5 group-hover:text-white"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L6.832 19.82a4.5 4.5 0 01-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 011.13-1.897L16.863 4.487zm0 0L19.5 7.125"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </a>
              <button
                class="flex items-center text-sm font-md bg-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-2 py-1 rounded-md false group bg-transparent rounded-md border hover:bg-red-400"
                data-testid="delete-5"
              >
                <svg
                  aria-hidden="true"
                  class="w-5 group-hover:text-white"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          class="odd:bg-white even:bg-gray-50 border-b hover:bg-gray-100"
        >
          <td
            class="px-6 py-4 font-medium text-gray-900 whitespace-nowrap"
          >
            <div
              class="flex items-center gap-3"
            >
              <img
                alt="NextJS logo"
                class="rounded-full"
                data-nimg="1"
                decoding="async"
                height="28"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fstatic-00.iconduck.com%2Fassets.00%2Fnext-js-icon-2048x2048-5dqjgeku.png&w=64&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fstatic-00.iconduck.com%2Fassets.00%2Fnext-js-icon-2048x2048-5dqjgeku.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fstatic-00.iconduck.com%2Fassets.00%2Fnext-js-icon-2048x2048-5dqjgeku.png&w=64&q=75 2x"
                style="color: transparent;"
                width="28"
              />
              <p>
                NextJS
              </p>
            </div>
          </td>
          <td
            class="px-6 py-4"
          >
            <div
              class="flex gap-3"
            >
              <a
                class="group rounded-md border p-2 hover:bg-blue-400"
                data-testid="edit-6"
                href="/edit-tech/6"
              >
                <svg
                  aria-hidden="true"
                  class="w-5 group-hover:text-white"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L6.832 19.82a4.5 4.5 0 01-1.897 1.13l-2.685.8.8-2.685a4.5 4.5 0 011.13-1.897L16.863 4.487zm0 0L19.5 7.125"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </a>
              <button
                class="flex items-center text-sm font-md bg-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-2 py-1 rounded-md false group bg-transparent rounded-md border hover:bg-red-400"
                data-testid="delete-6"
              >
                <svg
                  aria-hidden="true"
                  class="w-5 group-hover:text-white"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
